core/server/master/src/test/java/alluxio/master/metastore/rocks/RocksInodeStoreTest.java

```diff
@@ -14,42 +14,394 @@
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
 
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
 import alluxio.master.file.contexts.CreateDirectoryContext;
+import alluxio.master.file.meta.MutableInode;
 import alluxio.master.file.meta.MutableInodeDirectory;
 import alluxio.master.metastore.InodeStore.WriteBatch;
 
+import alluxio.master.metastore.ReadOption;
+import alluxio.resource.CloseableIterator;
+import alluxio.util.ThreadFactoryUtils;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadLocalRandom;
 
 public class RocksInodeStoreTest {
   @Rule
   public TemporaryFolder mFolder = new TemporaryFolder();
 
+  public RocksInodeStore mStore;
```

Comment:
I added some UT on how the ref count behaves correctly under some concurrency. @tcrain could you leave some comments on the test cases if you have some on top of your head? Also pls ignore how messy my code currently is. I intend to refactor code in this file once we settle on the test cases. Thx!

Right now all test cases here are passing.